src/odemis/acq/align/goffset.py

```
@@ -0,0 +1,805 @@
+                time.sleep(stabilization_time)
+
+            ccd_data = d.data.get(asap=False)
+            log_detector_state("GUI", "INITIAL", d, ccd_data)
+
+            if d is first_detector:
+                logging.info("Primary detector: using grating alignment (goffset)")
+                future._subfuture = (sparc_auto_grating_offset(spectrograph,d))
+
+            else:
+                logging.info("Secondary detector: using detector-offset alignment")
+                future._subfuture = (sparc_auto_detector_offset(spectrograph, d))
```

Ruff (0.15.14)

[error] Undefined name `sparc_auto_detector_offset`

(F821)